libs/exec/shell_execv.go

```diff
@@ -0,0 +1,38 @@
+package exec
+
+import "errors"
+
+func shellExecvOpts(content string, opts ExecvOptions) (ExecvOptions, error) {
+	if opts.Args != nil {
+		return ExecvOptions{}, errors.New("ShellExecv: Args is not supported")
+	}
+
+	shell, err := findShell()
+	if err != nil {
+		return ExecvOptions{}, err
+	}
+
+	ec, err := shell.prepare(content)
+	if err != nil {
+		return ExecvOptions{}, err
+	}
+
+	args := []string{ec.executable}
+	args = append(args, ec.args...)
+
+	return ExecvOptions{
```

Comment:
The input options are not carried over automatically. If this is intentional, then the `ShellExecv` command can/should take `dir` and `env` arguments and synthesize it directly. Then you can also get rid of the runtime check for non-empty `args`.

Reply:
Good point, done.